pkg/l2serviceattachmentparser/l2serviceattachmentparser.go

```diff
@@ -58,6 +66,41 @@ func (sattp *L2SrvAttParser) ParseL2ServiceAttachment(d *render.RenderData) (str
 		return "", err
 	}
 
+	if len(sattp.subnetResources) > 0 {
+		if len(sattp.subnetResources) > 2 {
+			err := errors.New("More than 2 subnets are not supported for a L2Service")
+			sattp.log.Error(err, "")
+		}
+		var ipType, ipamType string
+		for i, subnet := range sattp.subnetResources {
+			if i > 0 {
+				if ipType == subnet.Spec.Type {
+					err := errors.New("Subnets in one L2Service should not be of same ip type")
+					sattp.log.Error(err, "")
+					return "", err
+				}
+				if ipamType != subnet.Spec.Ipam {
+					err := errors.New("Subnets in one L2Service should have same ipam type")
+					sattp.log.Error(err, "")
+					return "", err
+				}
+			}
+			ipType = subnet.Spec.Type
+			ipamType = subnet.Spec.Ipam
+			sp := parser.NewSubnetParser(subnet, sattp.log)
+			if err := sp.ValidateSubnet(); err != nil {
+				return "", err
+			}
+			if err := sp.ValidateRoute(sattp.routeResources[subnet.GetName()]); err != nil {
+				return "", err
+			}
+		}
+		ipam := sattp.ipamMapping[ipamType]
+		cniConfigObj := sattp.instantiateIpamConfig()
```

Comment:
Can we change the cniConfigObj to ipamConfigObj so we do not mix it with the cnier Interface objects?

Reply:
Done.